crates/evm/src/decode.rs

```diff
@@ -26,52 +55,65 @@ pub fn decode_console_logs(logs: &[Log]) -> Vec<String> {
 /// This function returns [None] if it is not a DSTest log or the result of a Hardhat
 /// `console.log`.
 pub fn decode_console_log(log: &Log) -> Option<String> {
-    // NOTE: We need to do this conversion because ethers-rs does not
-    // support passing `Log`s
-    let raw_log = RawLog { topics: log.topics.clone(), data: log.data.to_vec() };
-    let decoded = match ConsoleEvents::decode_log(&raw_log).ok()? {
-        LogsFilter(inner) => format!("{}", inner.0),
-        LogBytesFilter(inner) => format!("{}", inner.0),
-        LogNamedAddressFilter(inner) => format!("{}: {:?}", inner.key, inner.val),
-        LogNamedBytes32Filter(inner) => {
-            format!("{}: {}", inner.key, B256::new(inner.val))
-        }
-        LogNamedDecimalIntFilter(inner) => {
-            let (sign, val) = inner.val.into_sign_and_abs();
-            format!(
-                "{}: {}{}",
-                inner.key,
-                sign,
-                ethers::utils::format_units(val, inner.decimals.as_u32()).unwrap()
-            )
-        }
-        LogNamedDecimalUintFilter(inner) => {
-            format!(
-                "{}: {}",
-                inner.key,
-                ethers::utils::format_units(inner.val, inner.decimals.as_u32()).unwrap()
-            )
-        }
-        LogNamedIntFilter(inner) => format!("{}: {:?}", inner.key, inner.val),
-        LogNamedUintFilter(inner) => format!("{}: {:?}", inner.key, inner.val),
-        LogNamedBytesFilter(inner) => {
-            format!("{}: {}", inner.key, inner.val)
-        }
-        LogNamedStringFilter(inner) => format!("{}: {}", inner.key, inner.val),
-        LogNamedArray1Filter(inner) => format!("{}: {:?}", inner.key, inner.val),
-        LogNamedArray2Filter(inner) => format!("{}: {:?}", inner.key, inner.val),
-        LogNamedArray3Filter(inner) => format!("{}: {:?}", inner.key, inner.val),
+    let raw_log = AlloyLog::new_unchecked(log.topics.into_iter().map(|h| h.to_alloy()).collect_vec(), log.data.0.into());
+    let data = log_address::abi_decode_data(&raw_log.data, false);
+    if let Ok(inner) = log::abi_decode_data(&raw_log.data, false) {
+        return Some(inner.0)
+    } else if let Ok(inner) = logs::abi_decode_data(&raw_log.data, false) {
+        return Some(format!("{}", Bytes::from(inner.0)))
+    } else if let Ok(inner) = log_address::abi_decode_data(&raw_log.data, false) {
```

Comment:
cc @DaniPopes there should be a better way to do this?

Reply:
Chatted with @DaniPopes about this, this feature is still missing from Alloy and needs to be implemented. We can leave this on abigen for now, and we'll come back to this once the feature is added, as it's only a single function